ml-agents/mlagents/trainers/ppo/trainer.py

```diff
@@ -91,6 +91,8 @@ def process_experiences(
         :param new_info: Dictionary of all next brains and corresponding BrainInfo.
         """
         info = new_info[self.brain_name]
+        if self.is_training:
```

Comment:
Does this crash if vector_observations is []?

Reply:
It should not.  Firstly because it should be impossible to use vector observations and not return any. Second of all because the tensor is of variable length.